vault/vault.go

```diff
@@ -130,6 +137,21 @@ func New(
 			return nil, err
 		}
 	}
+
+	if cd := utils.GetVaultParam(secretConfig, VaultCooldownPeriod); cd != "" {
+		confCooldownPeriod, err = time.ParseDuration(cd)
+		if err == nil && confCooldownPeriod > time.Minute {
+			logrus.Infof("cooldown period is set to %s via %s=%s", confCooldownPeriod, VaultCooldownPeriod, cd)
+		} else {
+			// let's turn OFF the cooldown, if one passed invalid value (or "disabled")
+			logrus.WithError(err).Warnf("cooldown period is invalid (%s=%s) -- cooldowns turned OFF", VaultCooldownPeriod, cd)
+			confCooldownPeriod = 0
+		}
```

Comment:
Lets error out here if the provided value is invalid. Only if someone provides == 0, then disable it.

Reply:
[148196b](https://github.com/libopenstorage/secrets/pull/86/commits/148196bffcdc2a86436036cb65868b72f3248ebb)